res/controllers/Denon-MC7000-scripts.js

```diff
@@ -0,0 +1,839 @@
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ * 
+ * Before using the mapping please make sure your MC7000 controller works for
+ * your operating system. For Windows you need driver software by Denon, Mac users 
+ * should be lucky as it shall work out-of-the-box. Linux users need to compile 
+ * their own Kernel with modified /sound/usb/clock.c see the "Denon MC7000 Mapping"
+ * thread at https://www.mixxx.org/forums/viewtopic.php?f=7&t=13126
+**/
+
+var MC7000 = {};
+
+/*///////////////////////////////////
+//      USER VARIABLES BEGIN       //
+///////////////////////////////////*/
+
+// Wanna have Needle Search active while playing a track ?
+// In any case Needle Search is available holding "SHIFT" down.
+// can be true or false (recommended: false)
+MC7000.needleSearchPlay = false;
+
+// Pitch Fader ranges to cycle through with the "RANGE" buttons.
+var lowest = 4   ;// lowest value in % (default: 4)
+var low = 6      ;// next value in % (default: 6)
+var middle = 10  ;// next value in % (default: 10)
+var high = 16    ;// next value in % (default: 16)
+var highest = 24 ;// highest value in % (default: 24)
+
+// Platter Ring LED mode
+// Mode 0 = Single "off" LED chase (all others "on")
+// Mode 1 = Single "on" LED chase (all others "off")
+// use "SHIFT" + "DECK #" to toggle between both modes
+MC7000.modeSingleLED = 1    ;// default: 1
+
+// Set Vinyl Mode on ("true") or off ("false") when MIXXX starts.
+// This sets the Jog Wheel touch detection / Vinyl Mode
+// and the Jog LEDs ("VINYL" on = spinny, "VINYL" off = track position).
+MC7000.VinylModeOn = true    ;// default: true
+
+// Scratch algorithm parameters
+MC7000.scratchParams = {
+    recordSpeed: 33.3   ,// default: 33.3
+    alpha: (1.0/10)     ,// default: (1.0/10)
+    beta: (1.0/10)/32    // default: (1.0/10)/32
+};
+
+// Sensitivity of the jog wheel (also depends on audio latency)
+// lower values make it less, higher value more sensible
+MC7000.jogParams = {
+    jogSensitivity: 30  ,// default: 30
+    maxJogValue: 3      ,// default: 3
+};
+
+/*/////////////////////////////////
+//      USER VARIABLES END       //
+/////////////////////////////////*/
+
+
+/* OTHER VARIABLES - DONT'T TOUCH EXCEPT YOU KNOW WHAT YOU DO */
+
+// Resolution of the jog wheel, set so the spinny
+// Jog LED to match exactly the movement of the Jog Wheel
+// The physical resolution seams to be around 1100
+MC7000.jogWheelTicksPerRevolution = 894;
```

Comment:
I did first start to count the numbers of lines generated when turning the jog wheel 5 rounds and divided by 5 to get to this 1100 ticks per round. Anyhow, if in scratch mode then the led around the platter moves quicker when moving the wheel back and forth. With a value of 894 the LED position actually stays aligned with the finger position on the platter.

Reply:
But the Javascript itself is responsible for the LED ring, not the controllers firmware, right?
I think the issue might not be the input resolution, but rather how you are scaling the output value.
Moreover, just counting the messages itself is probably not super accurate for determining the resolution, because controllers usually encode the speed of the platter in the message to save bandwidth. So you actually have to "record" all the values sent from those five rotations, sum them and then divide them again. Otherwise the resulting values will be inaccurate at higher speeds.
Instead of trying to match the platter to the controller LEDs, can you try to match the platter to the spinnies in the skin?

Reply:
OK. I made it in a similar way as described within the little video on Youtube https://www.youtube.com/watch?v=QUT8tZmwM00 which was linked from the WIKI page or the forum (can't remember where it was) in order to get scratch wheels going. Did it in the same way counting the ticks as from minute 6:16 in the video. I recognized though that some MIDI values become different when moving the wheel quicker so I turned it slowly to stay at the same MIDI values. The Platter , LEDs and GUI spinnies are aligned.